static/app/components/comboBox/index.tsx

```diff
@@ -0,0 +1,416 @@
+    isDismissable: true,
+    onInteractOutside: () => {
+      state.close();
+      inputRef.current?.blur();
+    },
+    shouldCloseOnBlur: true,
+  });
+
+  return (
+    <SelectContext.Provider
+      value={{
+        ...selectContext,
+        overlayIsOpen: state.isOpen,
+      }}
+    >
+      <ControlWrapper className={className}>
+        <StyledInput
+          {...inputProps}
+          placeholder={placeholder}
+          ref={mergeRefs([inputRef, triggerProps.ref])}
+          size={size}
+        />
+        <SizingDiv aria-hidden ref={sizingRef} size={size}>
+          {state.inputValue}
+        </SizingDiv>
+        <StyledPositionWrapper
+          {...overlayProps}
+          zIndex={theme.zIndex?.tooltip}
+          visible={state.isOpen}
+        >
+          <StyledOverlay ref={popoverRef}>
+            {isLoading && (
+              <MenuHeader size={size}>
+                <MenuTitle>{t('Loading...')}</MenuTitle>
+                <MenuHeaderTrailingItems>
+                  {isLoading && <StyledLoadingIndicator size={12} mini />}
+                </MenuHeaderTrailingItems>
+              </MenuHeader>
+            )}
+            <ListBox
+              {...listBoxProps}
+              ref={listBoxRef}
+              listState={state}
+              keyDownHandler={() => true}
+              size={size}
+              sizeLimitMessage={sizeLimitMessage}
+            />
+            <EmptyMessage>No items found</EmptyMessage>
+          </StyledOverlay>
+        </StyledPositionWrapper>
+      </ControlWrapper>
+    </SelectContext.Provider>
+  );
+}
+
+const getOptionTextValue = (option: ComboBoxOption<string>) => {
+  if (option.textValue) {
+    return option.textValue;
+  }
+  if (typeof option.label === 'string') {
+    return option.label;
+  }
+  return '';
+};
+
+/**
+ * Component that allows users to select an option from a dropdown list
+ * by typing in a input field
+ *
+ * **WARNING: This component is still experimental and may change in the future.**
+ */
+function ControlledComboBox<Value extends string>({
+  options,
+  sizeLimit,
+  value,
+  ...props
+}: Omit<ComboBoxProps<Value>, 'items' | 'defaultItems' | 'children'> & {
+  options: ComboBoxOptionOrSection<Value>[];
+  defaultValue?: Value;
+  onChange?: (value: ComboBoxOption<Value>) => void;
+  value?: Value;
+}) {
+  const [isFiltering, setIsFiltering] = useState(true);
+  const [inputValue, setInputValue] = useState(() => {
+    return (
+      options
+        .flatMap(item => ('options' in item ? item.options : [item]))
+        .find(option => option.value === value)?.label ?? ''
+    );
+  });
+
+  // Sync input value with value prop
+  const previousValue = useRef(value);
+  if (previousValue.current !== value) {
+    const selectedLabel = options
+      .flatMap(item => ('options' in item ? item.options : [item]))
+      .find(option => option.value === value)?.label;
+    if (selectedLabel) {
+      setInputValue(selectedLabel);
```

Comment:
setState inside render function

Reply:
Afaik this is the recommended approach for updating state from props:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Did we have problems with that?

Reply:
when is it used? The parent component is trying to filter things for you and then you can also edit the filter?

Reply:
No, the parent component controls the selected value and I need to sync the inputs text with it, when it changes.